### PR TITLE
Fixed issue [security] #19118: Improper permission management on bulk actions

### PR DESCRIPTION
**Pull request by Shnoulle**
Dev: fix send reset email : $userManager->canEdit()
Dev: fix Permission userManager->canAssignPermissions()
Dev: fix roles : Permission::model()->hasGlobalPermission('superadmin', 'create')
Dev: no need Permission on group : already done

### CODE REVIEW

File: application/controllers/UserManagementController.php [hunk 1]
```diff
@@ -1018,21 +1018,23 @@ public function actionBatchSendAndResetLoginData()
...
-                    if (!$success) {
-                        $aResults[$user]['error'] = sprintf(gT("Error: New password could not be sent to %s"), $oUser->email);
-                    }
-                    $aResults[$user]['result'] = $success;
-                }
+                continue;
+            }
+            $userManager = new UserManager(Yii::app()->user, $oUser);
+            if (!$userManager->canEdit()) {
+                $success = false;
```
Comment (gabrieljenik) [inline application/controllers/UserManagementController.php:1027]: or `$aResults[$user]['result'] = false` ?
I haven't analyzed it, but as there is a continue right after...

**Review by gabrieljenik — APPROVED** [review]
Code structure looks OK.
Left a comment on something which could be nothing or it may need amendment.

File: application/controllers/UserManagementController.php (hunk 1 above)
Comment (Shnoulle) [reply application/controllers/UserManagementController.php:1027]: You're rigth : `$aResults[$user]['result'] = false;`

**Review by gabrieljenik — APPROVED** [review]
Code structure looks good.

File: application/controllers/UserManagementController.php [hunk 2]
```diff
@@ -1066,8 +1068,22 @@ public function actionBatchPermissions()
...
+            ];
+            if ($oUser->uid == Yii::app()->user->id) {
+                $aResults[$user]['result'] = false;
+                $aResults[$user]['error'] = gT("You can not update your own permission.");
+                continue;
+            }
+            $userManager = new UserManager(Yii::app()->user, $oUser);
+            if (!$userManager->canAssignPermissions()) {
+                $results[$iUserId]['result'] = false;
+                $results[$iUserId]['error'] = gT("You do not have permission to this user.");
```
Comment (tiborpacalat) [inline application/controllers/UserManagementController.php:1083]: This should be "You are not allowed to assign permissions to this user."

**Review by tiborpacalat — CHANGES_REQUESTED** [review]
Can you please change the sentence to what I suggested? Thanks

**Pull request merged**